src/main/java/com/powsybl/openloadflow/ac/AcloadFlowEngine.java

```diff
@@ -78,7 +78,7 @@ private void runOuterLoop(OuterLoop outerLoop, OuterLoopContextImpl outerLoopCon
                 LOGGER.debug("Start outer loop '{}' iteration {}", outerLoop.getType(), runningContext.outerLoopTotalIterations);
 
                 // if not yet stable, restart Newton-Raphson
-                runningContext.lastNrResult = newtonRaphson.run(new PreviousValueVoltageInitializer());
+                runningContext.lastNrResult = newtonRaphson.run(new PreviousValueVoltageInitializer(), context.getNetwork().getReporter(), outerLoopIteration.toInteger() + 1, outerLoop.getType());
```

Comment:
We cannot do that! Separation of concern is a key of the design. NR does not known anything about outer loops.